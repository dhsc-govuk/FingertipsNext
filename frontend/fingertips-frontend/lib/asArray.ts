export const asArray = (value?: string | string[]): string[] => {
  if (value) {
    return Array.isArray(value) ? value : [value];
  }
  return [];
};
