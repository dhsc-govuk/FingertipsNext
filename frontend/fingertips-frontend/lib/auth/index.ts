import { AuthConfigFactory } from '@/lib/auth/config';
import NextAuth from 'next-auth';

export const { handlers, signIn, signOut, auth } = NextAuth(
  AuthConfigFactory.getConfig()
);
