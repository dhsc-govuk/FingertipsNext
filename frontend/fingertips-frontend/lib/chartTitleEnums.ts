export enum ChartTitlesEnum {
  LineChart = 'Indicator trends over time',
  BarChartEmbeddedTable = 'Compare areas for one time period',
  Heatmap = 'Overview of indicators and areas',
  SpineChart = 'Area profile by indicators',
  InequalitiesBarChart = 'Inequalities comparison for one time period',
  InequalitiesLineChart = 'Inequalities trends over time',
  ThematicMap = 'Compare an indicator by areas',
  PopulationPyramid = 'Population pyramid',
}
