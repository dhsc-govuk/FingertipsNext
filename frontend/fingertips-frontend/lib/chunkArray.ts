export const maxNumAreasThatCanBeRequestedAPI = 300;

export function chunkArray(
  arrayToChunk: string[],
  chunkSize: number = maxNumAreasThatCanBeRequestedAPI
) {
  const chunkedArray = [];
  for (let i = 0; i < arrayToChunk.length; i += chunkSize) {
    chunkedArray.push(arrayToChunk.slice(i, i + chunkSize));
  }
  return chunkedArray;
}
