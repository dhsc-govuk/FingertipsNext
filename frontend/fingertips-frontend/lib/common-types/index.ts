export enum Trend {
  INCREASING = 'Increasing',
  DECREASING = 'Decreasing',
  NO_SIGNIFICANT_CHANGE = 'No significant change',
  NOT_AVAILABLE = 'No recent trend data available',
}

export enum TrendCondition {
  GETTING_BETTER = 'getting better',
  GETTING_WORSE = 'getting worse',
}

export enum Direction {
  UP = 'up',
  DOWN = 'down',
  RIGHT = 'right',
  LEFT = 'left',
}
