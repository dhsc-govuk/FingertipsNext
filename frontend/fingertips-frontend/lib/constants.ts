export const siteTitle = 'Find public health data';

export const siteDescription = `${siteTitle} is a rich source of indicators across a range of health and wellbeing themes designed to support JSNA and commissioning to improve health and wellbeing, and reduce inequalities.`;

export const spaceSeparatedPattern = /^\d+(\s+\d+)+$/;
