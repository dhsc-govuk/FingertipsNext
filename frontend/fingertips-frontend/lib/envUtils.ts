import { env } from 'process';
import { EnvironmentVariables } from '@/EnvironmentVariables';

function missingVar(varName: string): never {
  throw new Error(`Env var '${varName}' needs to be defined`);
}

export function readEnvVar(varName: EnvironmentVariables): string {
  return env[varName] ?? missingVar(varName);
}

export function tryReadEnvVar(
  varName: EnvironmentVariables
): string | undefined {
  // Explictly returning undefined to block an empty string. This keeps the behaviour equivalent to readEnvVar.
  return env[varName] || undefined;
}
