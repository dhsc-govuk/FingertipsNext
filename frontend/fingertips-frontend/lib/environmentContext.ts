import { getEnvironmentVariable } from './utils';

export const DHSC_AI_SEARCH_USE_MOCK_SERVICE: string =
  'DHSC_AI_SEARCH_USE_MOCK_SERVICE';
export const DHSC_AI_SEARCH_SERVICE_URL: string = 'DHSC_AI_SEARCH_SERVICE_URL';
export const DHSC_AI_SEARCH_INDEX_NAME: string = 'DHSC_AI_SEARCH_INDEX_NAME';
export const DHSC_AI_SEARCH_API_KEY: string = 'DHSC_AI_SEARCH_API_KEY';
export const DHSC_AI_SEARCH_SCORING_PROFILE: string =
  'DHSC_AI_SEARCH_SCORING_PROFILE';

const loadEnvironmentContext = (): Map<string, string> => {
  console.log('Loading environment variables ...');
  const map: Map<string, string> = new Map();

  // Set required environment variables
  const mockService = getEnvironmentVariable(
    DHSC_AI_SEARCH_USE_MOCK_SERVICE
  ) as string;
  map.set(DHSC_AI_SEARCH_USE_MOCK_SERVICE, mockService);
  const useMock: boolean = mockService === 'true';

  // Following attributes are only mandatory if the mock search service is not being used.
  map.set(
    DHSC_AI_SEARCH_SERVICE_URL,
    getEnvironmentVariable(DHSC_AI_SEARCH_SERVICE_URL, !useMock) as string
  );
  map.set(
    DHSC_AI_SEARCH_INDEX_NAME,
    getEnvironmentVariable(DHSC_AI_SEARCH_INDEX_NAME, !useMock) as string
  );
  map.set(
    DHSC_AI_SEARCH_API_KEY,
    getEnvironmentVariable(DHSC_AI_SEARCH_API_KEY, !useMock) as string
  );

  // Set optional environment variables
  map.set(
    DHSC_AI_SEARCH_SCORING_PROFILE,
    getEnvironmentVariable(DHSC_AI_SEARCH_SCORING_PROFILE, false) as string
  );

  return map;
};

export const EnvironmentContext = (function () {
  let environmentMap: Map<string, string> | undefined;

  return {
    getEnvironmentMap: function () {
      if (!environmentMap) {
        environmentMap = loadEnvironmentContext();
      }
      return environmentMap;
    },
    reset: function () {
      environmentMap = undefined;
    },
  };
})();
