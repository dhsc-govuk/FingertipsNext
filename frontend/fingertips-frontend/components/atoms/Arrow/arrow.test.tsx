import { render, screen } from '@testing-library/react';
import { Arrow } from './index';
import { Direction } from '@/lib/common-types';

// filepath: frontend/fingertips-frontend/components/atoms/Arrow/index.test.tsx

describe('Arrow Component', () => {
  it('snapshot test', () => {
    const container = render(<Arrow direction={Direction.UP} />);

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders correctly with default props', () => {
    render(<Arrow direction={Direction.UP} />);
    const arrowIcon = screen.getByTestId('arrow-icon');
    expect(arrowIcon).toBeInTheDocument();
    expect(arrowIcon).toHaveAttribute('stroke', '#000000'); // Default stroke color
  });

  it('applies the correct rotation for Direction.UP', () => {
    render(<Arrow direction={Direction.UP} />);
    const arrowGroup = screen.getByTestId('arrow-up').closest('g');
    expect(arrowGroup).toHaveAttribute('transform', 'rotate(0)');
  });

  it('applies the correct rotation for Direction.RIGHT', () => {
    render(<Arrow direction={Direction.RIGHT} />);
    const arrowGroup = screen.getByTestId('arrow-right').closest('g');
    expect(arrowGroup).toHaveAttribute('transform', 'rotate(90 10 10)');
  });

  it('applies the correct rotation for Direction.DOWN', () => {
    render(<Arrow direction={Direction.DOWN} />);
    const arrowGroup = screen.getByTestId('arrow-down').closest('g');
    expect(arrowGroup).toHaveAttribute('transform', 'rotate(180 10 10)');
  });

  it('applies the correct rotation for Direction.LEFT', () => {
    render(<Arrow direction={Direction.LEFT} />);
    const arrowGroup = screen.getByTestId('arrow-left').closest('g');
    expect(arrowGroup).toHaveAttribute('transform', 'rotate(-90 10 10)');
  });

  it('applies the correct stroke color when strokeColour is provided', () => {
    const customColor = '#FF5733';
    render(<Arrow direction={Direction.UP} strokeColour={customColor} />);
    const arrowIcon = screen.getByTestId('arrow-icon');
    expect(arrowIcon).toHaveAttribute('stroke', customColor);
  });
});
