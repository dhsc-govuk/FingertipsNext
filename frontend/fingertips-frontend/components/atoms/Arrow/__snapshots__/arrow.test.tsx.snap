// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Component snapshot test 1`] = `
<DocumentFragment>
  <svg
    data-testid="arrow-icon"
    fill="none"
    height="15"
    stroke="#000000"
    stroke-linecap="square"
    stroke-width="1.5"
    viewBox="0 0 20 20"
    width="15"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      transform="rotate(0)"
    >
      <path
        d="M12 19V6M5 12l7-7 7 7"
        data-testid="arrow-up"
      />
    </g>
  </svg>
</DocumentFragment>
`;
